tests/functional/idempotency/test_idempotency.py

```diff
@@ -0,0 +1,586 @@
+import copy
+import sys
+
+import pytest
+from botocore import stub
+
+from aws_lambda_powertools.utilities.idempotency.exceptions import (
+    IdempotencyAlreadyInProgressError,
+    IdempotencyInconsistentStateError,
+    IdempotencyPersistenceLayerError,
+    IdempotencyValidationError,
+)
+from aws_lambda_powertools.utilities.idempotency.idempotency import idempotent
+from aws_lambda_powertools.utilities.validation import envelopes, validator
+
+TABLE_NAME = "TEST_TABLE"
+
+
```

Comment:
Maybe add a missing test for the invalid status returned from the data store

```suggestion

def test_data_record_invalid_status_value():
    data_record = DataRecord("key", status="UNSUPPORTED_STATUS")

    with pytest.raises(IdempotencyInvalidStatusError) as e:
        data_record.status

    assert e.value.args[0] == "UNSUPPORTED_STATUS"
```